src/main/cpp/_nix_based/jssc.cpp

```diff
@@ -524,28 +534,90 @@ JNIEXPORT jboolean JNICALL Java_jssc_SerialNativeInterface_writeBytes
     return result == bufferSize ? JNI_TRUE : JNI_FALSE;
 }
 
+/**
+ * Waits until 'read()' has something to tell for the specified filedescriptor.
+ */
+static void awaitReadReady(JNIEnv*env, jlong fd){
+#if HAVE_POLL == 0
+    // Alternative impl using 'select' as 'poll' isn't available (or broken).
+
+    //assert(fd < FD_SETSIZE); // <- Might help when hunting SEGFAULTs.
+    fd_set readFds;
+    while(true) {
+        FD_ZERO(&readFds);
+        FD_SET(fd, &readFds);
+        int result = select(fd + 1, &readFds, NULL, NULL, NULL);
+        if(result < 0){
+            // man select: On error, -1 is returned, and errno is set to indicate the error
+            // TODO: Maybe a candidate to raise a java exception. But won't do
+            //       yet for backward compatibility.
+            continue;
+        }
+        // Did wait successfully.
+        break;
+    }
+    FD_CLR(fd, &readFds);
+
+#else
+    // Default impl using 'poll'. This is more robust against fd>=1024 (eg
+    // SEGFAULT problems).
+
+    struct pollfd fds[1];
+    fds[0].fd = fd;
+    fds[0].events = POLLIN;
+    while(true){
+        int result = poll(fds, 1, -1);
+        if(result < 0){
+            // man poll: On error, -1 is returned, and errno is set to indicate the error.
```

Comment:
won't error be returned in case the port has been closed? what would happen if we call `readBytes` and then `closePort`? won't this just continue indefinitely?

Reply:
AFAIK this ends up in an infinite-loop exactly as the currently released code does (by [ignoring return code of *select()*](https://github.com/java-native/jssc/blob/v2.9.2/src/main/cpp/_nix_based/jssc.cpp#L540)). So its at least "backward compatible" ;)

Further there's also an infinite-loop in case of EOF because code currently just loops if "*read() < 0*" (See [code of v2.9.2](https://github.com/java-native/jssc/blob/v2.9.2/src/main/cpp/_nix_based/jssc.cpp#L542)). (It's easily reproducible: Just pull the usb cable while reading and observe what happens with the CPU)

I feel there is a lot of error-handling missing everywhere in native code. But I decided against trying to fix everything in a single PR at once because:
- Tends to end up in messy (as in *hard-to-review*) PRs.
- Will bring breaking changes in most cases.

Maybe I should open individual issues to document all my findings (currently only visible here and there in my scratch-book).

WDYT?

Reply:
I'm hoping for a palatable solution. 

> this ends up in an infinite-loop exactly as the currently released code does

Should we just merge as-is and handle the exceptions at a later time?

Reply:
Checking in to see if we're in agreement on merging this?

Reply:
Sure, I don't mind. There's still `Thread.sleep` mentioned before this, which I don't think is needed anymore though. But I'm fine merging as-is

Reply:
> there's still `Thread.sleep` mentioned before this, which I don't think is needed anymore though. But I'm fine merging as-is

It was factored out but GitHub for some reason didn't put the `outdated` keyword on the message.  It also doesn't allow me to reply or resolve it for some reason.  From what I can see, it's removed from the PR.

Reply:
Whoops, it's still there.  https://github.com/java-native/jssc/blob/1e9b04ebe53c0724245def5bd10d92a190aeeebf/src/main/java/jssc/SerialPort.java#L559.  @hiddenalpha, thoughts?

Reply:
Sleep has been disabled, no issues encountered as far as higher CPU loading after applying monitor lock to act in its place globally in SerialPort.java

Reply:
Sleep removed per 1aff5f9.